### PR TITLE
Feature/survey email notification

### PR DESCRIPTION
**Pull request by XanderVertegaal**
This PR mainly consists of commits you have seen before, with a few extras (based on your PR comments):

- A few template infelicities (missing closing tags etc.) have been fixed.
- Debug flags are now dependent on the presence and value of a `DEVELOPMENT` variable in .env. The README has been updated to reflect this.
- Doctests have been added for `is_fill_correct()` and to theck the contents of the to-be-sent email. Unittest is used to check that the email is sent asynchronously via Celery.

I have asked Shalom whether we can use coloringbook@uu.nl as a 'sender' email address. I will update this PR accordingly once I know more.

The only thing I did not manage was to find a way to not hardcode `'config.py'` in `make_celery.py`. Feel free to add your own ideas or commits if you know how to accomplish this.

### CODE REVIEW

File: coloringbook/templates/email/email.html [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        <tr>
+            <td>{{ evaluation.page }}</td>
+            <td>{{ evaluation.target }}</td>
+            <td>{{ evaluation.color }}</td>
+            <td>{{ evaluation.correct }}</td>
+        </tr>
+        {% else %}
+            <tr>
+                <td colspan="3">Er zijn geen resultaten geregistreerd.</td>
+            </tr>
```
Comment (jgonggrijp) [inline coloringbook/templates/email/email.html:31]: I suspect you have one more level of indentation here than you intended.

File: coloringbook/utilities.py [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+            )
+            raise
+    return actions
+
+
+def is_fill_correct(fill):
+    """
+    Returns whether a fill action is correct.
+
+    `fill` is expected to a Fill object.
```
Comment (jgonggrijp) [inline coloringbook/utilities.py:206]:
```suggestion
    `fill` is expected to be a Fill object.
```

File: coloringbook/utilities.py [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+    ...     incorrect_result = is_fill_correct(test_incorrect_fill)
+
+    # Result if fill matches expectation
+    >>> correct_result
+    True
+
+    # Result if fill does not match expectation
+    >>> incorrect_result
+    False
+    """
```
Comment (jgonggrijp) [inline coloringbook/utilities.py:278]: My compliments for the clarity of these tests!

**Review by jgonggrijp — APPROVED** [review]
Excellent job, well done! :+1: 

> The only thing I did not manage was to find a way to not hardcode `'config.py'` in `make_celery.py`. Feel free to add your own ideas or commits if you know how to accomplish this.

[`sys.argv`](https://docs.python.org/3/library/sys.html#sys.argv). Or if you want to take it all the way, [`argparse`](https://docs.python.org/3/library/argparse.html).

File: coloringbook/templates/email/email.html (hunk 1 above)
Comment (XanderVertegaal) [reply coloringbook/templates/email/email.html:31]: Quite so! Well noticed.

File: coloringbook/mail/utilities.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    ...     first_message = outbox[0]
+
+    >>> outbox_length
+    1
+
+    >>> first_message.subject
+    'ColoringBook - nieuwe resultaten opgeslagen'
+
+    >>> first_message.html
+    u'<p>Hallo,</p>\\n\\n<p>Dit is een automatisch gegenereerde e-mail van Coloring Book.</p>\\n\\n<p>De vragenlijst test is ingevuld door 1 deelnemer(s).</p>\\n    \\n<p>De resultaten zijn als volgt:</p>\\n\\n\\n<table>\\n    <caption>Bob (2000-01-01)</caption>\\n    <thead>\\n        <tr>\\n            <th>Pagina</th>\\n            <th>Doel</th>\\n            <th>Kleur</th>\\n            <th>Goed/fout</th>\\n        </tr>\\n    </thead>\\n    <tbody>\\n        \\n        <tr>\\n            <td>testpage</td>\\n            <td>left door</td>\\n            <td>black</td>\\n            <td>Fout</td>\\n        </tr>\\n        \\n        <tr>\\n            <td>testpage</td>\\n            <td>left door</td>\\n            <td>white</td>\\n            <td>Fout</td>\\n        </tr>\\n        \\n    </tbody>\\n</table>\\n\\n\\n<p>Hartelijke groeten,</p>\\n\\n<p>Het Coloring Bookteam</p>'
```
[the remark this answers is not in the document]
Comment (XanderVertegaal) [reply coloringbook/mail/utilities.py:98]: Ah yes, I see what you mean. Good idea! I'll happily take it on board for future tests. That should make it easier to change the styling/layout of a template without messing up tests like these that are aimed at its contents.

**Comment by XanderVertegaal** [comment]
> [`sys.argv`](https://docs.python.org/3/library/sys.html#sys.argv). Or if you want to take it all the way, [`argparse`](https://docs.python.org/3/library/argparse.html).

Using sys.argv turned out to be a bit difficult, since `make_celery.py` (which needs the argument) is run from a different container. I ended up adding the name of the configuration file to `.env`, so I can use it in both the `app` container and the `celery` container.

**Comment by jgonggrijp** [comment]
To the contrary, I already expected the name of the `config.py` to be in the `.env`, so I think you took the right approach!

**Pull request merged**